src/Analyser/LazyScopeFactory.php

```diff
@@ -101,7 +98,7 @@ public function create(
 			$currentlyAllowedUndefinedExpressions,
 			$nativeExpressionTypes,
 			$inFunctionCallsStack,
-			$this->treatPhpDocTypesAsCertain,
+			true,
```

Comment:
`treatPhpDocTypesAsCertain` of MutatingScope would only be changed by `doNotTreatPhpDocTypesAsCertain`

Reply:
so not
> Basically it says "don't use PHPDocs". Which isn't what the setting is about.

The meaning of the setting is kept and when to use doNotTreatPhpDocTypesAsCertain is handled by the rule side.